src/DataStructures/VectorImpl.hpp

```diff
@@ -639,15 +641,101 @@ template <typename T>
+// appropriately recursive. This form primarily works by using templates to
+// ensure that our comparison operator is resolved with higher priority that the
+// blaze form as of blaze 3.8
+template <
+    typename Lhs, typename Rhs,
+    Requires<(is_derived_of_vector_impl_v<Lhs> or
+              is_derived_of_vector_impl_v<
+                  Rhs>)and not(std::is_base_of_v<blaze::Computation, Lhs> or
+                               std::is_base_of_v<blaze::Computation, Rhs>) and
+             not(std::is_same_v<Rhs, typename Lhs::ElementType> or
+                 std::is_same_v<Lhs, typename Rhs::ElementType>)> = nullptr>
+bool operator==(const Lhs& lhs, const Rhs& rhs) noexcept {
+  return blaze::equal<blaze::strict>(lhs, rhs);
+}
+
+template <
+    typename Lhs, typename Rhs,
+    Requires<(is_derived_of_vector_impl_v<Lhs> or
+              is_derived_of_vector_impl_v<
+                  Rhs>)and not(std::is_base_of_v<blaze::Computation, Lhs> or
+                               std::is_base_of_v<blaze::Computation, Rhs>) and
+             not(std::is_same_v<Rhs, typename Lhs::ElementType> or
+                 std::is_same_v<Lhs, typename Lhs::ElementType>)> = nullptr>
+bool operator!=(const Lhs& lhs, const Rhs& rhs) noexcept {
+  return not(lhs == rhs);
+}
+
+// Impose strict equality for any expression templates; note that
+// this overrides intended behavior in blaze for comparison
+// operators to use approximate equality in favor of equality
+// between containers being appropriately recursive. This form
+// primarily works by using templates to ensure that our
+// comparison operator is resolved with higher priority that the
+// blaze form as of blaze 3.8
+template <typename Lhs, typename Rhs,
+          Requires<std::is_base_of_v<blaze::Computation, Lhs> or
+                   std::is_base_of_v<blaze::Computation, Rhs>> = nullptr>
+bool operator==(const Lhs& lhs, const Rhs& rhs) noexcept {
+  return blaze::equal<blaze::strict>(lhs, rhs);
+}
+
+template <typename Lhs, typename Rhs,
+          Requires<std::is_base_of_v<blaze::Computation, Lhs> or
+                   std::is_base_of_v<blaze::Computation, Rhs>> = nullptr>
+bool operator!=(const Lhs& lhs, const Rhs& rhs) noexcept {
+  return not(lhs == rhs);
+}
+
+template <typename Lhs, Requires<is_derived_of_vector_impl_v<Lhs>> = nullptr>
+bool operator==(const Lhs& lhs,
+                const typename Lhs::ElementType& rhs) noexcept {
+  for (const auto& element : lhs) {
+    if (element != rhs) {
+      return false;
+    }
+  }
+  return true;
+}
+
+template <typename Lhs, Requires<is_derived_of_vector_impl_v<Lhs>> = nullptr>
+bool operator!=(const Lhs& lhs,
+                const typename Lhs::ElementType& rhs) noexcept {
+  return not(lhs == rhs);
+}
+
+template <typename Rhs, Requires<is_derived_of_vector_impl_v<Rhs>> = nullptr>
+bool operator==(const typename Rhs::ElementType& lhs,
+                const Rhs& rhs) noexcept {
+  for (const auto& element : lhs) {
```

Comment:
This function and the one with the elements swapped both loop over `lhs`.  I don't understand this enough to know if that is correct.

Maybe just implement this as `rhs == lhs` in any case.